lms/djangoapps/grades/api/tests/test_views.py

```diff
@@ -245,3 +250,228 @@ def test_grade(self, grade, result):
         }
         expected_data.update(result)
         self.assertEqual(resp.data, [expected_data])  # pylint: disable=no-member
+
+
+class GradingPolicyTestMixin(object):
+    """
+    Mixin class for Grading Policy tests
+    """
+    view = None
+
+    def setUp(self):
+        super(GradingPolicyTestMixin, self).setUp()
+        self.create_user_and_access_token()
+
+    def create_user_and_access_token(self):
+        # pylint: disable=missing-docstring
+        self.user = GlobalStaffFactory.create()
+        self.oauth_client = ClientFactory.create()
+        self.access_token = AccessTokenFactory.create(user=self.user, client=self.oauth_client).token
+
+    @classmethod
+    def create_course_data(cls):
+        # pylint: disable=missing-docstring
+        cls.invalid_course_id = 'foo/bar/baz'
+        cls.course = CourseFactory.create(display_name='An Introduction to API Testing', raw_grader=cls.raw_grader)
+        cls.course_id = unicode(cls.course.id)
+        with cls.store.bulk_operations(cls.course.id, emit_signals=False):
+            cls.sequential = ItemFactory.create(
+                category="sequential",
+                parent_location=cls.course.location,
+                display_name="Lesson 1",
+                format="Homework",
+                graded=True
+            )
+
+            factory = MultipleChoiceResponseXMLFactory()
+            args = {'choices': [False, True, False]}
+            problem_xml = factory.build_xml(**args)
+            cls.problem = ItemFactory.create(
+                category="problem",
+                parent_location=cls.sequential.location,
+                display_name="Problem 1",
+                format="Homework",
+                data=problem_xml,
+            )
+
+            cls.video = ItemFactory.create(
+                category="video",
+                parent_location=cls.sequential.location,
+                display_name="Video 1",
+            )
+
+            cls.html = ItemFactory.create(
+                category="html",
+                parent_location=cls.sequential.location,
+                display_name="HTML 1",
+            )
+
+        cls.empty_course = CourseFactory.create(
+            start=datetime(2014, 6, 16, 14, 30),
+            end=datetime(2015, 1, 16),
+            org="MTD",
+            # Use mongo so that we can get a test with a SlashSeparatedCourseKey
+            default_store=ModuleStoreEnum.Type.mongo
+        )
+
+    def http_get(self, uri, **headers):
+        """
+        Submit an HTTP GET request
+        """
+
+        default_headers = {
+            'HTTP_AUTHORIZATION': 'Bearer ' + self.access_token
+        }
+        default_headers.update(headers)
+
+        response = self.client.get(uri, follow=True, **default_headers)
+        return response
+
+    def http_get_for_course(self, course_id=None, **headers):
```

Comment:
Looks like every call to this function is immediately followed by a line of the form `self.assertEqual(response.status_code, {status code})`. Maybe bring that assert into this function and rename it for the sake of DRY?